tests/test_factory_auto_registration.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+from pydantic import BaseModel
+
+from pydantic_factories import ModelFactory
+
+
+class A(BaseModel):
+    a_text: str
+
+
+class B(BaseModel):
+    b_text: str
+    a: A
+
+
+class C(BaseModel):
+    b: B
+    b_list: List[B]
+
+
+def test_auto_register_model_factory() -> None:
```

Comment:
Could you please also add a test that would work with a typed dict and a dataclass as a model?

I think your code should work fine for that case but it is great to be sure that future refactorings won't break the code.